source/solvers/gd_navier_stokes.cc

```diff
@@ -1215,7 +1215,17 @@ GDNavierStokesSolver<dim>::solve()
       this->dynamic_flow_control();
 
       if (this->simulation_control->is_at_start())
-        this->iterate();
+        {
+          for (unsigned int i = 0;
+               i <
+               this->simulation_parameters.mesh_adaptation.initial_refinement;
+               i++)
+            NavierStokesBase<dim,
+                             TrilinosWrappers::MPI::BlockVector,
+                             std::vector<IndexSet>>::refine_mesh();
+
+          this->iterate();
```

Comment:
This is a good idea really, but I suggest you move it to the initial condition. This way, for example, you could re-apply the initial condition between every mesh refinement. In addition, this way every solver would have this capability automatically.
I really suggest moving it to initial conditions in the NavierStokesBase and to reapply the initial condition every time.